app/src/behaviors/behavior_hold_tap.c

```diff
@@ -217,16 +230,67 @@ static void clear_hold_tap(struct active_hold_tap *hold_tap) {
     hold_tap->work_is_cancelled = false;
 }
 
-static void decide_balanced(struct active_hold_tap *hold_tap, enum decision_moment event) {
-    switch (event) {
+static bool
+does_position_of_first_other_keypress_permit_hold_behavior(struct active_hold_tap *hold_tap) {
+    if (hold_tap->config->hold_enabler_keys_len == 0) {
+        return true;
+    }
+    for (int i = 0; i < hold_tap->config->hold_enabler_keys_len; i++) {
+        if (hold_tap->config->hold_enabler_keys[i] ==
+            hold_tap->position_of_first_other_key_pressed) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static void store_position_if_is_first_other_key_pressed(
+    struct active_hold_tap *active_hold_tap,
+    struct decision_trigger_event *decision_trigger_event) {
+    // The value of position_of_first_other_key_pressed is initialized to -1 when
+    // on_hold_tap_binding_pressed calls store_hold_tap.
+    // If position_of_first_other_key_pressed is >= 0, then this is not the first
+    // other key to have been pressed.
+    if (active_hold_tap->position_of_first_other_key_pressed >= 0) {
+        return;
+    }
+
+    // If the decision_moment is not an other-key-down event, nothing to do.
+    if (decision_trigger_event->decision_moment != HT_OTHER_KEY_DOWN) {
+        return;
+    }
+
+    // If the position of the key being pressed is the hold-tap key, something went wrong.
+    if (active_hold_tap->position == decision_trigger_event->other_key_position) {
+        LOG_DBG("ERROR hold-tap key cannot be pressed while still active");
+        return;
+    }
+
+    // All checks passed. Store the position of the other key in the active_hold_tap.
+    active_hold_tap->position_of_first_other_key_pressed =
+        decision_trigger_event->other_key_position;
+    return;
+}
+
+static void decide_balanced(struct active_hold_tap *hold_tap,
+                            struct decision_trigger_event *decision_trigger_event) {
+    switch (decision_trigger_event->decision_moment) {
     case HT_KEY_UP:
         hold_tap->status = STATUS_TAP;
         return;
     case HT_OTHER_KEY_UP:
-        hold_tap->status = STATUS_HOLD_INTERRUPT;
+        if (does_position_of_first_other_keypress_permit_hold_behavior(hold_tap)) {
+            hold_tap->status = STATUS_HOLD_INTERRUPT;
+        } else {
+            hold_tap->status = STATUS_TAP;
+        }
         return;
     case HT_TIMER_EVENT:
-        hold_tap->status = STATUS_HOLD_TIMER;
+        if (does_position_of_first_other_keypress_permit_hold_behavior(hold_tap)) {
```

Comment:
We don't need to check this on the timer event, because the 'other keypress event' will already have set `STATUS_HOLD_INTERRUPT` if `does_position_of_first_other_keypress_permit_hold_behavior` is True.

(thus, for this check the global storage of `hold_tap->position_of_first_other_key_pressed` is not required either)